src/vm/peimage.inl

```diff
@@ -481,8 +481,19 @@ inline BOOL PEImage::HasNativeHeader()
         return GetLoadedLayout()->HasNativeHeader();
     else
     {
-        PEImageLayoutHolder pLayout(GetLayout(PEImageLayout::LAYOUT_ANY,LAYOUT_CREATEIFNEEDED));
```

Comment:
Yes it is thrown during the call BinderHasNativeHeader above.  We could move the try catch block there.  I placed it in PEImage::HasNativeHeader() as that seemed like the most general location to place this.  Let me know what @gkhanna79  and @janvorli prefer.